engine/src/main/java/org/hibernate/validator/messageinterpolation/ExpressionLanguageFeatureLevel.java

```diff
@@ -0,0 +1,49 @@
+/*
+ * Hibernate Validator, declare and validate application constraints
+ *
+ * License: Apache License, Version 2.0
+ * See the license.txt file in the root directory or <http://www.apache.org/licenses/LICENSE-2.0>.
+ */
+package org.hibernate.validator.messageinterpolation;
+
+import org.hibernate.validator.Incubating;
+import org.hibernate.validator.constraintvalidation.HibernateConstraintValidatorContext;
+
+/**
+ * Indicates the level of features enabled for the Expression Language engine.
+ *
+ * @since 6.2
+ */
+@Incubating
+public enum ExpressionLanguageFeatureLevel {
+
+	/**
+	 * Expression Language expressions are not interpolated.
+	 */
+	NONE,
+
+	/**
+	 * Only allows access to the variables injected via
+	 * {@link HibernateConstraintValidatorContext#addExpressionVariable(String, Object)}, the Jakarta Bean
+	 * Validation-defined {@code formatter} and the {@code ResourceBundle}s.
+	 */
+	SAFEST,
+
+	/**
+	 * Only allows to access bean properties and the Jakarta Bean Validation-defined {@code formatter}.
+	 * <p>
+	 * This is the minimal level to have a specification-compliant implementation.
+	 */
+	BEAN_PROPERTIES,
+
+	/**
+	 * This level allows bean method execution and can lead to serious security issues if not very carefully handled.
+	 * <p>
+	 * If using this level, you need to be sure you are not injecting user input in an expression without properly
+	 * escaping it using {@link HibernateConstraintValidatorContext#addExpressionVariable(String, Object)}.
+	 */
+	INSECURE;
+
+	public static final ExpressionLanguageFeatureLevel DEFAULT = ExpressionLanguageFeatureLevel.BEAN_PROPERTIES;
```

Comment:
That's pretty neat, but will it not lead to confusion when calling `enableExpressionLanguage(ExpressionLanguageFeatureLevel.DEFAULT)`? One can wonder if you're resetting to the application-configured defaults or to the HV defaults.

Alternatively you could add a `DEFAULT` value to the enum, and handle it wherever necessary.

Reply:
That's a good idea.